### PR TITLE
chore: make Embed and AppStore Tests required

### PR DESCRIPTION
**Pull request by hariombalhara**
## What does this PR do?

Make all tests required now. 
A recent regression that could have been avoided if embed tests were required #10357

Side Note: Marked it High Priority as it's quick to merge and super important to have these tests block PR merge considering recent regression as well.

### CODE REVIEW

File: .github/workflows/pr.yml [hunk 1]
```diff
@@ -84,21 +84,18 @@ jobs:
...
   e2e-embed:
     name: E2E embeds tests
-    if: ${{ needs.changes.outputs.embed == 'true' }}
     needs: [changes, lint, build]
     uses: ./.github/workflows/e2e-embed.yml
     secrets: inherit
 
   e2e-embed-react:
     name: E2E React embeds tests
-    if: ${{ needs.changes.outputs.embed-react == 'true' }}
```
Comment (hariombalhara) [inline .github/workflows/pr.yml:101]: Now there is no condition to run these checks.

File: .github/workflows/pr.yml [hunk 2]
```diff
@@ -109,7 +106,7 @@ jobs:
     secrets: inherit
 
   required:
-    needs: [lint, type-check, test, build, e2e]
+    needs: [lint, type-check, test, build, e2e, e2e-embed, e2e-embed-react, e2e-app-store]
```
Comment (hariombalhara) [inline .github/workflows/pr.yml:109]: As there is no condition on these checks, we can require them here.

**Review by keithwillcode — CHANGES_REQUESTED** [review]
@hariombalhara to avoid adding more minutes of wait time back to builds, let’s implement the matrix strategy like we have on the main e2e test suite.

**Comment by hariombalhara** [comment]
@keithwillcode I have pushed the changes for matrix support. Yet to test them. But I noticed that the embed and AppStore tests don't take more time than core e2e tests.
AFAIK, All these tests run in parallel, so even without sharding they would compete much earlier than core tests.

If I am right in that analysis, we don't need sharding.

image

**Comment by keithwillcode** [comment]
IMG_7832

@hariombalhara 13 minutes is what they used to take before I put the matrix in place. Now they only take 2-3 minutes.

**Review by keithwillcode — APPROVED** [review]

**Pull request merged**